crates/engine/tree/src/tree/mod.rs

```diff
@@ -2294,7 +2303,20 @@ where
         // Extend with block we are validating root for.
         input.append_ref(hashed_state);
 
-        Ok(ParallelStateRoot::new(consistent_view, input).incremental_root_with_updates()?)
+        let (tx, rx) = std::sync::mpsc::channel();
+        let blocking_task_pool = self.blocking_task_pool.clone();
+
+        rayon::spawn(move || {
+            let result: Result<(B256, TrieUpdates), AsyncStateRootError> =
+                futures::executor::block_on(async move {
```

Comment:
this is a bit bit pointless because we spawn a rayon task and await the stateroot call

better approaches here would be to 
a) don't use async in the state root computation (sync channel)
b) use tokio block_on on the state root call

Reply:
cool thanks, finally went for option a), ptal